app/assets/javascripts/angular/services/graph_refresher.js

```diff
@@ -31,6 +26,14 @@ angular.module("Prometheus.services").factory('GraphRefresher',
               'data': data
             });
             break;
+          case 'vector':
+            var d = data.value || data.Value;
+            d.forEach(function(s) {
+              s.metric.serverName = server.name;
+              s.expressionID = expressionID;
```

Comment:
scopes the legend string to an expression. previously it was to the server, which you discovered caused a bug.